src/lib/drupal/drupalClient.ts

```diff
@@ -16,3 +17,16 @@ export const drupalClient = new DrupalClient(baseUrl, {
   // cache: redisCache(createRedisClient(process.env.REDIS_URL)),
   previewSecret: process.env.DRUPAL_PREVIEW_SECRET,
 })
+
+// Fetch drupal menu resource with cache
+export async function getMenu(name: string, params: QueryParams<null>) {
+  const menu = await drupalClient.getMenu(name, {
+    params: params().getQueryObject(),
+
+    // Cache resource during build, not dev.
+    // withCache: process.env.NEXT_PHASE === PHASE_PRODUCTION_BUILD,
+    // cacheKey: `menu:${name}`,
+  })
+
+  return menu
+}
```

Comment:
This move is interesting to me. Why `drupalClient`? It seems like the job of this file is to create an instance of `DrupalClient` which can then be used elsewhere. The previous structure made more sense to me, at least naively. `getMenu` feels like a drupal utility that belongs in `src/lib/drupal/[menu|query|etc].ts` OR in `src/data/queries/menu.ts`. Maybe I'm thinking of this incorrectly, though.

Reply:
no, i think you are thinking of it correctly and i was maybe a little overzealous moving stuff here. I was putting it in drupalClient because it's essentially a wrapper around `drupalClient.getMenu`, but it does make sense to leave `drupalClient.ts` as solely creating the DrupalClient instance, and `getMenu` is a utility that belongs elsewhere.